lr-cli/cortx_setup/commands/storage/controller_config.py

```diff
@@ -0,0 +1,177 @@
+#
+# Copyright (c) 2020 Seagate Technology LLC and/or its Affiliates
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU Affero General Public License as published
+# by the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+# GNU Affero General Public License for more details.
+# You should have received a copy of the GNU Affero General Public License
+# along with this program. If not, see <https://www.gnu.org/licenses/>.
+# For any questions about this software or licensing,
+# please email opensource@seagate.com or cortx-questions@seagate.com.
+#
+
+from pathlib import Path
+from ..command import Command
+from cortx.utils.conf_store import Conf
+from provisioner.commands import PillarSet
+from provisioner.salt import local_minion_id, function_run
+from .commons import Commons
+
+prvsnr_cluster_path = Path(
+    '/opt/seagate/cortx_configs/provisioner_cluster.json'
+)
+
+
+"""Cortx Setup API for configuring the storage controller"""
+
+class StorageControllerConfig(Command):
+    '''
+    $ cortx_setup storage controller --user <user> --password <password>
+
+    $ cortx_setup storage controller --type (Gallium | Indium | virtual)
+
+    $ cortx_setup storage controller --mode (primary | secondary) --ip <ip-address> --port <port-number>
+    '''
+    _args = {
+        'user': {
+            'type': str,
+            'default': None,
+            'optional': True,
+            'help': 'username for controller e.g {manage}'
+        },
+        'password': {
+            'type': str,
+            'default': None,
+            'optional': True,
+            'help': 'password for controller user'
+        },
+        'type': {
+            'type': str,
+            'optional': True,
+            'help': 'Type of controller e.g {Gallium | Indium | virtual}'
```

Comment:
Make sure all examples are in small case:
{gallium | indium | virtual}

Reply:
done